core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Profiler.scala

```diff
@@ -49,12 +49,29 @@ class Profiler(hadoopConf: Configuration, appArgs: ProfileArgs, enablePB: Boolea
 
   private val useAutoTuner: Boolean = appArgs.autoTuner()
   private var progressBar: Option[ConsoleProgressBar] = None
+  // Store application status reports indexed by event log path.
+  private val appStatusReporter = new ConcurrentHashMap[String, AppResult]
 
   override val outputDir = appArgs.outputDirectory().stripSuffix("/") +
     s"/${Profiler.SUBDIR}"
 
   logInfo(s"Threadpool size is $nThreads")
 
+  /**
+   * For each app status report, generate a StatusProfileResult.
+   * @return Seq[StatusProfileResult] - Seq[(path, status, description)]
+   */
+  private def generateStatusProfResults(appStatuses: Seq[AppResult]): Seq[StatusProfileResult] = {
+    appStatuses.map {
+      case FailureAppResult(path, message) => StatusProfileResult(path, "FAILURE", message)
+      case SkippedAppResult(path, message) => StatusProfileResult(path, "SKIPPED", message)
+      case SuccessAppResult(path, _, message) => StatusProfileResult(path, "SUCCESS", message)
+      case UnknownAppResult(path, _, message) => StatusProfileResult(path, "UNKNOWN", message)
+      case profAppResult: AppResult =>
+        throw new UnsupportedOperationException(s"Invalid status for $profAppResult")
+    }
+  }
+
```

Comment:
once you use same class StatusProfileResult for both Q/P tools, this code can be used by Qualification and P tools

Reply:
reuse `generateStatusProfResults` for both Q/P tool, moved definition to `Profiler.generateStatusProfResults`